src/daemon/jsonrpc/api.rs

```diff
@@ -364,43 +452,356 @@ impl RpcApi for RpcImpl {
+            (&emergency_tx, emer_sigs),
+            (&unvault_emergency_tx, unvault_emer_sigs),
+        ) {
+            return Err(JsonRpcError::invalid_params(format!(
+                "Error while sharing signatures: {}",
+                e
+            )));
         }
 
+        // NOTE: it will only mark it as 'securing' if it was 'funded', not if it was
+        // marked as 'secured' by db_update_presigned_tx() !
+        db_mark_securing_vault(&db_path, db_vault.id).map_err(|e| internal_error!(e))?;
+
         Ok(json!({}))
     }
 
-    fn listpresignedtransactions(
+    fn getunvaulttx(
         &self,
         meta: Self::Metadata,
-        outpoints: Option<Vec<OutPoint>>,
+        outpoint: OutPoint,
     ) -> jsonrpc_core::Result<serde_json::Value> {
-        let (response_tx, response_rx) = mpsc::sync_channel(0);
-        assume_ok!(
-            meta.tx.send(RpcMessageIn::ListPresignedTransactions(
-                outpoints,
-                response_tx,
-            )),
-            "Sending 'listpresignedtransactions' to main thread"
+        stakeholder_only!(meta);
+        let revaultd = meta.rpc_utils.revaultd.read().unwrap();
+        let db_file = &revaultd.db_file();
+
+        // We allow the call for Funded 'only' as unvaulttx would later fail if it's
+        // not 'secured'.
+        let vault = match db_vault_by_deposit(db_file, &outpoint).map_err(|e| internal_error!(e))? {
+            None => {
+                return Err(JsonRpcError::invalid_params(
+                    RpcControlError::UnknownOutpoint(outpoint).to_string(),
+                ));
+            }
+            Some(vault) => match vault.status {
+                VaultStatus::Unconfirmed => {
+                    return Err(JsonRpcError::invalid_params(
+                        RpcControlError::InvalidStatus((vault.status, VaultStatus::Funded))
+                            .to_string(),
+                    ));
+                }
+                _ => vault,
+            },
+        };
+
+        // Derive the descriptors needed to create the UnvaultTransaction
+        let deposit_descriptor = revaultd
+            .deposit_descriptor
+            .derive(vault.derivation_index, &revaultd.secp_ctx);
+        let deposit_txin = DepositTxIn::new(
+            outpoint,
+            DepositTxOut::new(vault.amount.as_sat(), &deposit_descriptor),
         );
-        let vaults = assume_ok!(
-            response_rx.recv(),
-            "Receiving 'listpresignedtransactions' from main thread"
+        let unvault_descriptor = revaultd
+            .unvault_descriptor
+            .derive(vault.derivation_index, &revaultd.secp_ctx);
+        let cpfp_descriptor = revaultd
+            .cpfp_descriptor
+            .derive(vault.derivation_index, &revaultd.secp_ctx);
+
+        let unvault_tx =
+            UnvaultTransaction::new(deposit_txin, &unvault_descriptor, &cpfp_descriptor, 0)
+                .map_err(|e| internal_error!(e))?;
```

Comment:
Unrelated but since you are touching this line already, can you:
```diff
diff --git a/src/daemon/jsonrpc/api.rs b/src/daemon/jsonrpc/api.rs
index 69456c4..8434ed1 100644
--- a/src/daemon/jsonrpc/api.rs
+++ b/src/daemon/jsonrpc/api.rs
@@ -658,7 +658,7 @@ impl RpcApi for RpcImpl {
             .derive(vault.derivation_index, &revaultd.secp_ctx);
 
         let unvault_tx =
-            UnvaultTransaction::new(deposit_txin, &unvault_descriptor, &cpfp_descriptor, 0)
+            UnvaultTransaction::new(deposit_txin, &unvault_descriptor, &cpfp_descriptor, revaultd.lock_time)
                 .map_err(|e| internal_error!(e))?;
 
         Ok(json!({
```